### PR TITLE
feat(exporter): add content-app download log exporter

### PR DESCRIPTION
**Pull request by decko**
## Summary

- New `export-content-logs` CLI command that exports Python (.whl/.whl.metadata) and RPM (.rpm) download logs from content-app (pulp-content) CloudWatch streams to Parquet
- Parses package metadata from filenames: wheel regex (from pip/pulp_python) for Python, NEVRA parser (from pulp_rpm) for RPM
- Type-specific Parquet schemas with common fields (timestamp, domain, distribution, package_name, package_version, architecture, artifact_size, cache_hit, org_id) plus Python-specific (build_tag, pyver, abi) or RPM-specific (epoch, release)
- Fixes timezone bug in existing `parse_time()` (utcnow → timezone-aware) and switches to `logGroupNames` (plural) for JSON-structured log support

## Test plan

- [x] 36 unit tests passing (31 new + 5 existing, no regressions)
- [x] Live-tested against prod CloudWatch: 4,138 Python downloads and 3,742 RPM downloads exported in 10 min window
- [ ] Deploy CronJob config (follow-up: PULP-1831)

Jira: [PULP-1811](https://redhat.atlassian.net/browse/PULP-1811)
Epic: [PULP-1809](https://redhat.atlassian.net/browse/PULP-1809)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[PULP-1811]: https://redhat.atlassian.net/browse/PULP-1811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PULP-1809]: https://redhat.atlassian.net/browse/PULP-1809?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by Sourcery

Add a new CLI command for exporting pulp-content download logs from CloudWatch to Parquet, with typed schemas for Python wheels and RPMs, and improve time handling and CloudWatch querying.

New Features:
- Introduce an export-content-logs CLI entrypoint to export pulp-content download logs for Python wheels and RPMs to Parquet with S3 support.
- Add parsing of content-app access log lines and content paths to extract package metadata and request details for Python and RPM artifacts.
- Define dedicated PyArrow schemas and conversion logic for Python and RPM download records.

Bug Fixes:
- Make parse_time return timezone-aware UTC datetimes instead of naive utcnow values.
- Switch CloudWatch Logs querying to use logGroupNames to support structured log groups.

Tests:
- Add unit tests covering content log parsing, filename parsing for wheels and RPMs, content-type filtering, schema-based Parquet conversion, and CloudWatch query building.
- Extend test fixtures with sample content CloudWatch results and Parquet paths for the new exporter.

### CODE REVIEW

**Comment by decko** [comment]
Addressed Sourcery review feedback in 5b3a6e2:

**Comment 1 (bug_risk — defensive timestamp parsing):** Fixed. `_parse_timestamp` now returns `None` for empty/malformed values; the record is skipped.

**Comment 2 (skip counters):** Fixed. Added 5 per-category counters (log parse, path parse, extension, malformed filename, bad timestamp) printed to stderr for operational visibility.

**Comment 3 (tests for malformed records):** Partially addressed. Added 3 tests: invalid/non-matching records, malformed filename warning (capsys), and bad timestamps. Skipped the suggested non-content-path test — the suggested implementation had a bug (`.replace("/pulp/content/")` but our prefix is `/api/pulp-content/`).

**Comment 4 (edge case tests for helpers):** Not addressing. The suggested implementation tries to `copy.deepcopy` a fixture and modify top-level keys like `edge_record["cache"]`, but these values are embedded inside the `message` string and parsed by regex — the test wouldn't work as written. The helper functions are 3 lines each and already covered by integration tests.

File: management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/content_cloudwatch.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+def _parse_org_id(org_value):
+    if not org_value or org_value == "-":
+        return None
+    return org_value
+
+
+def _parse_timestamp(timestamp_str):
+    if timestamp_str.endswith("Z"):
+        timestamp_str = timestamp_str[:-1]
+    return datetime.fromisoformat(timestamp_str)
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/content_cloudwatch.py]: Fixed in 5b3a6e2. `_parse_timestamp` now returns `None` for empty/malformed values, and the record is skipped.

File: management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/content_cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    schema = SCHEMAS[content_type]
+    filename_parser = FILENAME_PARSERS[content_type]
+    records = []
+    skipped_count = 0
+
+    for result in results:
+        message = result.get("message", result.get("@message", ""))
+        timestamp_str = result.get("@timestamp", "")
+
+        parsed_line = parse_content_log_line(message)
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/content_cloudwatch.py:81]: Fixed in 5b3a6e2. Added 5 per-category skip counters (log parse, path parse, extension, malformed filename, bad timestamp) printed to stderr for operational visibility.

File: management_tools/pulp-access-logs-exporter/tests/test_content_export.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+class TestContentToParquetPython:
+    def test_converts_python_downloads(self, sample_content_cloudwatch_results, sample_content_parquet_path):
+        table = convert_content_to_arrow_table(
+            sample_content_cloudwatch_results, "python"
+        )
+        assert table.num_rows == 2
+        assert table.schema == PYTHON_SCHEMA
+
+        write_parquet(table, sample_content_parquet_path)
+        read_table = pq.read_table(sample_content_parquet_path)
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/tests/test_content_export.py:195]: Partially addressed in 5b3a6e2. Added tests for: invalid/non-matching records, malformed filename warning (capsys), and bad timestamps. Skipped the suggested non-content-path test — the implementation had a bug (`.replace("/pulp/content/")` but our prefix is `/api/pulp-content/`).

File: management_tools/pulp-access-logs-exporter/tests/test_content_export.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+class TestContentToParquetRpm:
+    def test_converts_rpm_downloads(self, sample_content_cloudwatch_results, sample_content_parquet_path):
+        table = convert_content_to_arrow_table(
+            sample_content_cloudwatch_results, "rpm"
+        )
+        assert table.num_rows == 2
+        assert table.schema == RPM_SCHEMA
+
+        write_parquet(table, sample_content_parquet_path)
+        read_table = pq.read_table(sample_content_parquet_path)
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/tests/test_content_export.py:255]: Not addressing. The suggested implementation tries to `copy.deepcopy` a fixture and modify top-level keys like `edge_record["cache"]`, but these values are embedded inside the `message` string and parsed by regex — the test wouldn't work as written. The helpers are 3 lines each and already covered by integration tests.
